odoo/fields.py

```diff
@@ -986,7 +986,12 @@ def __get__(self, record, owner):
         if self.compute and (record.id in env.all.tocompute.get(self, ())) \
                 and not env.is_protected(self, record):
             # self must be computed on record
-            recs = record if self.recursive else env.records_to_compute(self)
+            if self.recursive:
+                recs = record
+            else:
+                # do not mix real records with new records
```

Comment:
comment should be above the filtered, and maybe should more clearly note that we only want to compute records with the same "newness" as the reference, as at first glance it kind-of gives the impression that we're *partitioning* the records